<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Dialog</class>
 <widget class="QDialog" name="Dialog">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>548</width>
    <height>450</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Dialog</string>
  </property>
  <layout class="QVBoxLayout" name="verticalLayout">
   <item>
    <widget class="QTabWidget" name="tabWidget">
     <property name="currentIndex">
      <number>3</number>
     </property>
     <widget class="QWidget" name="tab">
      <attribute name="title">
       <string>Сфера</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout">
       <item row="3" column="1">
        <widget class="QDoubleSpinBox" name="s_z"/>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_3">
         <property name="text">
          <string>x:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QDoubleSpinBox" name="s_x"/>
       </item>
       <item row="3" column="0">
        <widget class="QLabel" name="label_5">
         <property name="text">
          <string>z:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="4">
        <widget class="QDoubleSpinBox" name="s_r"/>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_4">
         <property name="text">
          <string>y:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="3">
        <widget class="QLabel" name="label_2">
         <property name="text">
          <string>Радиус:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QDoubleSpinBox" name="s_y"/>
       </item>
       <item row="0" column="0" colspan="2">
        <widget class="QLabel" name="label">
         <property name="text">
          <string>Центр сферы:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="2" rowspan="4">
        <widget class="Line" name="line_4">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_2">
      <attribute name="title">
       <string>Параллелипипед</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_2">
       <item row="1" column="3">
        <widget class="QLabel" name="label_8">
         <property name="text">
          <string>y:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QDoubleSpinBox" name="paral_z"/>
       </item>
       <item row="5" column="1">
        <widget class="QDoubleSpinBox" name="paral_w"/>
       </item>
       <item row="4" column="2" rowspan="2">
        <widget class="Line" name="line_8">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="4" column="3">
        <widget class="QLabel" name="label_12">
         <property name="text">
          <string>высота:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2" rowspan="2">
        <widget class="Line" name="line_7">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QDoubleSpinBox" name="paral_x"/>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_6">
         <property name="text">
          <string>x:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="1">
        <widget class="QDoubleSpinBox" name="paral_l"/>
       </item>
       <item row="4" column="4">
        <widget class="QDoubleSpinBox" name="paral_h"/>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_9">
         <property name="text">
          <string>z:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="4">
        <widget class="QLabel" name="label_7">
         <property name="text">
          <string>Координаты левой нижней вершины:</string>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_11">
         <property name="text">
          <string>ширина:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0">
        <widget class="QLabel" name="label_10">
         <property name="text">
          <string>длина:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="4">
        <widget class="QDoubleSpinBox" name="paral_y"/>
       </item>
       <item row="3" column="0" colspan="5">
        <widget class="Line" name="line">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="6" column="0" colspan="2">
        <widget class="QLabel" name="label_29">
         <property name="text">
          <string>угол поворота вокруг оси Оу:</string>
         </property>
        </widget>
       </item>
       <item row="6" column="3">
        <widget class="QDoubleSpinBox" name="paral_a"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_3">
      <attribute name="title">
       <string>Пирамида</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_3">
       <item row="2" column="0">
        <widget class="QLabel" name="label_14">
         <property name="text">
          <string>z:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QDoubleSpinBox" name="pyr_x"/>
       </item>
       <item row="3" column="0" colspan="5">
        <widget class="Line" name="line_2">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_18">
         <property name="text">
          <string>высота:</string>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="4">
        <widget class="QLabel" name="label_15">
         <property name="text">
          <string>Координаты левой нижней вершины:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0" colspan="2">
        <widget class="QLabel" name="label_19">
         <property name="text">
          <string>длина стороны основания:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="4">
        <widget class="QDoubleSpinBox" name="pyr_y"/>
       </item>
       <item row="1" column="3">
        <widget class="QLabel" name="label_17">
         <property name="text">
          <string>y:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_13">
         <property name="text">
          <string>x:</string>
         </property>
        </widget>
       </item>
       <item row="2" column="1">
        <widget class="QDoubleSpinBox" name="pyr_z"/>
       </item>
       <item row="1" column="2" rowspan="2">
        <widget class="Line" name="line_5">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="6" column="0" colspan="2">
        <widget class="QLabel" name="label_30">
         <property name="text">
          <string>угол поворота вокруг оси Оу:</string>
         </property>
        </widget>
       </item>
       <item row="6" column="3">
        <widget class="QDoubleSpinBox" name="pyr_a"/>
       </item>
       <item row="5" column="3">
        <widget class="QDoubleSpinBox" name="pyr_h"/>
       </item>
       <item row="4" column="3">
        <widget class="QDoubleSpinBox" name="pyr_l"/>
       </item>
      </layout>
     </widget>
     <widget class="QWidget" name="tab_4">
      <attribute name="title">
       <string>Призма</string>
      </attribute>
      <layout class="QGridLayout" name="gridLayout_4">
       <item row="3" column="0" colspan="5">
        <widget class="Line" name="line_3">
         <property name="orientation">
          <enum>Qt::Horizontal</enum>
         </property>
        </widget>
       </item>
       <item row="0" column="0" colspan="5">
        <widget class="QLabel" name="label_16">
         <property name="text">
          <string>Координаты левой нижней вершины:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="0">
        <widget class="QLabel" name="label_21">
         <property name="text">
          <string>x:</string>
         </property>
        </widget>
       </item>
       <item row="4" column="0" colspan="2">
        <widget class="QLabel" name="label_24">
         <property name="text">
          <string>длина стороны основания:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="3">
        <widget class="QLabel" name="label_22">
         <property name="text">
          <string>y:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="4">
        <widget class="QDoubleSpinBox" name="pr_y"/>
       </item>
       <item row="5" column="0">
        <widget class="QLabel" name="label_20">
         <property name="text">
          <string>высота:</string>
         </property>
        </widget>
       </item>
       <item row="1" column="2" rowspan="2">
        <widget class="Line" name="line_6">
         <property name="orientation">
          <enum>Qt::Vertical</enum>
         </property>
        </widget>
       </item>
       <item row="1" column="1">
        <widget class="QDoubleSpinBox" name="pr_x"/>
       </item>
       <item row="2" column="1">
        <widget class="QDoubleSpinBox" name="pr_z"/>
       </item>
       <item row="2" column="0">
        <widget class="QLabel" name="label_23">
         <property name="text">
          <string>z:</string>
         </property>
        </widget>
       </item>
       <item row="6" column="0" colspan="2">
        <widget class="QLabel" name="label_31">
         <property name="text">
          <string>угол поворота вокруг оси Оу:</string>
         </property>
        </widget>
       </item>
       <item row="6" column="3">
        <widget class="QDoubleSpinBox" name="pr_a"/>
       </item>
       <item row="5" column="3">
        <widget class="QDoubleSpinBox" name="pr_h"/>
       </item>
       <item row="4" column="3">
        <widget class="QDoubleSpinBox" name="pr_l"/>
       </item>
      </layout>
     </widget>
    </widget>
   </item>
   <item>
    <widget class="QGroupBox" name="groupBox">
     <property name="title">
      <string>Материал</string>
     </property>
     <layout class="QFormLayout" name="formLayout">
      <item row="1" column="0">
       <widget class="QLabel" name="label_25">
        <property name="text">
         <string>цвет:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="0">
       <widget class="QLabel" name="label_26">
        <property name="text">
         <string>зеркальность:</string>
        </property>
       </widget>
      </item>
      <item row="3" column="1">
       <widget class="QDoubleSpinBox" name="spec">
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <widget class="QLabel" name="label_27">
        <property name="text">
         <string>коэффициент отражения: </string>
        </property>
       </widget>
      </item>
      <item row="4" column="1">
       <widget class="QDoubleSpinBox" name="refl">
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="5" column="0">
       <widget class="QLabel" name="label_28">
        <property name="text">
         <string>шереховатость</string>
        </property>
       </widget>
      </item>
      <item row="5" column="1">
       <widget class="QDoubleSpinBox" name="rough">
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>16777215</height>
         </size>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="pushButton">
        <property name="minimumSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <property name="maximumSize">
         <size>
          <width>30</width>
          <height>30</height>
         </size>
        </property>
        <property name="styleSheet">
         <string notr="true">background-color: rgb(255, 0, 0);
border-radius: 8px;</string>
        </property>
        <property name="text">
         <string/>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
   <item>
    <widget class="QDialogButtonBox" name="buttonBox">
     <property name="font">
      <font>
       <weight>75</weight>
       <bold>true</bold>
      </font>
     </property>
     <property name="orientation">
      <enum>Qt::Horizontal</enum>
     </property>
     <property name="standardButtons">
      <set>QDialogButtonBox::Cancel|QDialogButtonBox::Ok</set>
     </property>
    </widget>
   </item>
  </layout>
 </widget>
 <resources/>
 <connections>
  <connection>
   <sender>buttonBox</sender>
   <signal>accepted()</signal>
   <receiver>Dialog</receiver>
   <slot>accept()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>227</x>
     <y>430</y>
    </hint>
    <hint type="destinationlabel">
     <x>157</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>buttonBox</sender>
   <signal>rejected()</signal>
   <receiver>Dialog</receiver>
   <slot>reject()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>295</x>
     <y>436</y>
    </hint>
    <hint type="destinationlabel">
     <x>286</x>
     <y>274</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>pushButton</sender>
   <signal>clicked()</signal>
   <receiver>Dialog</receiver>
   <slot>color()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>233</x>
     <y>269</y>
    </hint>
    <hint type="destinationlabel">
     <x>607</x>
     <y>244</y>
    </hint>
   </hints>
  </connection>
 </connections>
 <slots>
  <slot>color()</slot>
 </slots>
</ui>
